src/charm.py

```diff
@@ -113,14 +113,16 @@ def __init__(self, framework: ops.Framework):
             self.on.delete_stale_profiles_action, self._on_delete_stale_profiles
         )
 
-    def _on_event_sync_profiles(self, event: ops.EventBase):
+    def _on_event_sync_profiles(self, _: ops.EventBase):
         """Update the Profiles if we can connect to the workload container."""
         try:
             self._sync_profiles()
-        except (ApiError, InvalidKfamAnnotationsError) as e:
-            logger.error(f"Could not sync profiles, due to the following error: {str(e)}")
         except ErrorWithStatus as e:
             logger.error(f"Could not sync profiles, due to the following error: {e.msg}")
+            self.unit.status = e.status
+        except Exception as e:
+            logger.error(f"Could not sync profiles, due to the following error: {str(e)}")
+            self.unit.status = ops.BlockedStatus("")
```

Comment:
suggestion: Although we are catching most exceptions, I'm not sure I like having a Blocked unit with zero explanation. Maybe adding a simple "Please look at the charm's logs for more details" could be more helpful as a message.
```suggestion
            self.unit.status = ops.BlockedStatus("Unexpected error, please look at the charm's logs for more details.")
```